src/components/Icon/Icon.tsx

```diff
@@ -0,0 +1,10 @@
+interface IconProps {
```

Comment:
You should have multiple icons not like this for example `PlusIcon`, `MinusIcon`, each icon has different html tag. For example: 
```tsx
export const ArrowIcon = ({
  className = 'size-4 text-heading',
}: ArrowIconProps) => (
  <svg width="12" height="12" viewBox="0 0 12 12" className={className}>
    <path
      d="M1 11L11 1M11 1H1M11 1V11"
      stroke="currentColor"
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
);
```